src/internal/storage/fileset/buffer.go

```diff
@@ -25,52 +33,75 @@ func NewBuffer() *Buffer {
 	}
 }
 
-func (b *Buffer) Add(path, datum string) io.Writer {
+func (b *Buffer) addInternal(path, datum string) *file {
```

Comment:
I think just `add` is fine. It being unexported implicitly means it is internal.